drivers/include/disp_dev.h

```diff
@@ -42,22 +44,28 @@ extern "C" {
  */
 typedef struct disp_dev disp_dev_t;
 
+/**
+ * @brief Type to store the update coordinates in
+ */
+typedef struct {
+    uint16_t x1; /**< Left coordinate */
+    uint16_t x2; /**< Right coordinate */
+    uint16_t y1; /**< Top coordinate */
+    uint16_t y2; /**< Bottom coordinate */
+} disp_dev_coordinates_t;
+
```

Comment:
There's no need to change the driver API. Just the disp_dev API needs the be changed and the ili9341 adapter functions can handle the conversion.
This will keep this PR minimal and avoid an API change (for something that is not marked as experimental, the ili9341 driver).

Reply:
Having 6 arguments for a function is bad practice. As I remember clean code is referring to a maximum of 3/4.
And we can easily reduce the amount of arguments here.

Reply:
Okay, fair enough. But why should I touch this as well? So I can reduce the amount of arguments there as well? To get to the ideal of 3/4 arguments for a function.

Reply:
> why shouldn't I touch this as well? So I can reduce the amount of arguments there as well?

The PR title says that it touches the disp_dev API, the underlying display driver doesn't have to be changed. This keeps the PR focused on the disp_dev changes and this eases and speeds up the reviews.

Reply:
> Okay, fair enough. But why shouldn't I touch this as well? So I can reduce the amount of arguments there as well? To get to the ideal of 3/4 arguments for a function.

@Citrullin could you split the ili9341 changes into another PR? It can be based on this one, that way everyone is happy (I think :) )